iothub_client/inc/iothub_device_client_ll.h

```diff
@@ -374,6 +374,106 @@ typedef struct IOTHUB_CLIENT_CORE_LL_HANDLE_DATA_TAG* IOTHUB_DEVICE_CLIENT_LL_HA
+*                                            result of the transaction.
+* @param[in]    userContextCallback          Optional User specified context that will be provided to the callback.
+*
+* @remarks   The application behavior is undefined if the user calls
+*            the IoTHubDeviceClient_LL_Destroy function from within any callback.
+*
+* @return    IOTHUB_CLIENT_OK upon success or an error code upon failure.
+*/
+IOTHUB_CLIENT_RESULT IoTHubDeviceClient_LL_SendPropertiesAsync(
+                         IOTHUB_DEVICE_CLIENT_LL_HANDLE iotHubClientHandle, 
+                         const unsigned char* properties,
+                         size_t propertiesLength,
+                         IOTHUB_CLIENT_PROPERTY_ACKNOWLEDGED_CALLBACK propertyAcknowledgedCallback,
+                         void* userContextCallback);
+/**
+* @brief   Retrieves all properties from IoT Hub.
+*
+* @param[in]   iotHubClientHandle    The handle created by a call to the create function.
+* @param[in]   propertyCallback      Callback invoked when properties are retrieved.
+                                     The API IoTHubClient_Deserialize_Properties can help deserialize the raw 
+*                                    payload stream.
+* @param[in]   userContextCallback   Optional user specified context that will be provided to the
+*                                    callback.
+*
+* @remarks  The application behavior is undefined if the user calls
+*           the IoTHubDeviceClient_LL_Destroy function from within any callback.
+*
+* @return   IOTHUB_CLIENT_OK upon success or an error code upon failure.
+*/
+IOTHUB_CLIENT_RESULT IoTHubDeviceClient_LL_GetPropertiesAsync(
+                        IOTHUB_DEVICE_CLIENT_LL_HANDLE iotHubClientHandle, 
+                        IOTHUB_CLIENT_PROPERTIES_RECEIVED_CALLBACK propertyCallback, 
+                        void* userContextCallback);
+
+/**
+* @brief   Retrieves all properties from IoT Hub and also subscribes for updates to writable properties.
+*
+* @param[in]   iotHubClientHandle      The handle created by a call to the create function.
+* @param[in]   propertyUpdateCallback  Callback both on initial retrieval of properties stored on IoT Hub
+                                       and subsequent service-initiated modifications of writable properties.
+                                       The API IoTHubClient_Deserialize_Properties can help deserialize the raw 
+*                                      payload stream.
+* @param[in]   userContextCallback     Optional user specified context that will be provided to the callback.
+*
+* @remarks  The application behavior is undefined if the user calls
+*           the IoTHubDeviceClient_LL_Destroy function from within any callback.
+*
+* @return   IOTHUB_CLIENT_OK upon success or an error code upon failure.
+*/
+IOTHUB_CLIENT_RESULT IoTHubDeviceClient_LL_GetPropertiesAndSubscribeToUpdatesAsync(
```

Comment:
nit: can the wording of this API be reworded? 
```suggestion
IOTHUB_CLIENT_RESULT IoTHubDeviceClient_LL_SubscribeToUpdatesAndGetPropertiesAsync(
```
Because if I understand correctly here the subscription is sync and then the get properties part is async? Just fits the call sequence better.